src/firecracker/examples/uffd/valid_handler.rs

```diff
@@ -53,15 +40,13 @@ fn main() {
         // We expect to receive either a Page Fault or Removed
         // event (if the balloon device is enabled).
         match event {
-            userfaultfd::Event::Pagefault { addr, .. } => {
-                uffd_handler.serve_pf(addr as *mut u8, len)
-            }
+            userfaultfd::Event::Pagefault { addr, .. } => uffd_handler.serve_pf(addr.cast(), len),
             userfaultfd::Event::Remove { start, end } => uffd_handler.update_mem_state_mappings(
-                start as *mut u8 as u64,
-                end as *mut u8 as u64,
+                start.cast::<u8>() as u64,
+                end.cast::<u8>() as u64,
```

Comment:
Nit: I don't think you need to cast the pointer type.
```
assert_eq!(x.cast::<A>() as u64, x.cast::<B>() as u64);
```
will always be true regardless of the types of `A` or `B`.

(unless for some reason the compiler doesn't allow `as u64` for certain types)

Reply:
done